src/modules/automata.rs

```diff
@@ -99,20 +100,35 @@ impl Automata {
     }
 
     pub fn full_match(&self, expr: &str) -> bool {
-        if let Some(matched) = self.greedy_search_impl(expr, true) {
-            matched.len() == expr.len()
-        } else {
-            false
+        match self.greedy_search(expr) {
+            Ok(result) => {
+                if let Some(matched) = result {
+                    matched.len() == expr.len()
+                } else {
+                    false
+                }
+            }
+            Err(error) => {
+                println!("Warning: {error:#?}");
```

Comment:
I think we should avoid swallowing an error and printing it to stdout. I would pass it up to the caller.

(Or have greedy_search not return a Result, and just let tests handle it.)

Reply:
I think this is outdated, since I reverted back to what we have before (`greedy_search` returns `Option<String>`).